update qc section of recipe

ErinWeisbart opened the pull request:
Updates QC section including:

Major changes:
- Plot empty cells per well separately from cell quality ratios per well
- Overhaul and simplification of correlation metrics to single by well plot
- Changing how some column names are determined so that they are filtered from lists of actual column names

Minor changes:
- Make wells square on by-well plots to better mimic whole-well-stitched images
- Changes to plot text aesthetics for readability
- More descriptive plot labels

gwaygenomics commented on 0.preprocess-sites/4.image-and-segmentation-qc.py:
```diff
@@ -164,31 +166,30 @@
     .rename(columns={0: "Ratio"})
 )
-
+empty_df=empty_df[['Percent_Empty']].stack().to_frame().reset_index().rename(columns={0: "Percent Empty"})
```
I am not sure how else you are using `empty_df`, but if possible, you should try to keep data processing of a single variable all in one spot. What I mean is move this line closer to where you define `empty_df` (line 155).

This will help reader understanding a lot.

gwaygenomics commented on 0.preprocess-sites/4.image-and-segmentation-qc.py:
```diff
@@ -164,31 +166,30 @@
-    + gg.coord_fixed()
+    + gg.geom_point(gg.aes(fill="Ratio"), shape='s', size=6)
+    + gg.geom_text(gg.aes(label="site_location"), size=6, color="lightgrey")
+    + gg.facet_wrap("~cell_quality_recode + well", ncol = 3, scales="fixed")
```
please make sure to run `black` on this script.

(Black will make sure there are no spaces in `ncol=3` 😄  )

ErinWeisbart replied:
hahaha. I can't even count the number of times I told myself "run `black` before making the PR" and yet I still forgot!

gwaygenomics commented on 0.preprocess-sites/4.image-and-segmentation-qc.py:
```diff
@@ -164,31 +166,30 @@
         axis_text=gg.element_blank(),
         axis_title=gg.element_blank(),
-        strip_background=gg.element_rect(colour="black", fill="#fdfff4"),
+        strip_background=gg.element_rect(colour="black", fill="#fdfff4", height = 1/5),
```
black will fix this too

gwaygenomics commented on 0.preprocess-sites/4.image-and-segmentation-qc.py:
```diff
@@ -291,42 +347,40 @@
 pll_col_prefix = "ImageQuality_PowerLogLogSlope_"
 PLLS_df_cols = image_meta_col_list.copy()
-PLLS_cols = []
-for name in painting_image_names:
```
Just confirming that you no longer create a figure for all `painting_image_names`. Instead you are creating a single figure, correct?

ErinWeisbart replied:
I don't think `painting_image_names` was ever used to make separate figures? Yes, this is a single figure for all the painting images.

gwaygenomics reviewed (approved):
Looks great to me. I made only minor comments that probably could use addressing before you merge. Thanks!

ErinWeisbart commented:
@gwaygenomics Did a little more cleanup.
- addressed your comments
- fixed error in how percent empty figure was plotting and saving
- made similar changes to saturation plot as other plots to change column name determination
- saturation plot: made channels categorical and reordered them so that the wrap keeps A, C, G, T in columns (moving DAPI to end)

gwaygenomics commented:
very nice - please merge when you are ready. (I assume you have write access)
